src/color.c

```diff
@@ -676,6 +750,18 @@ struct colors *parse_colors(void) {
 		goto fail;
 	}
 
+#if __FreeBSD__
+	char *force_env = getenv("CLICOLOR_FORCE");
+	char *clicolor = getenv("CLICOLOR");
+	char *colorterm = getenv("COLORTERM");
+	if ((force_env && (strcmp(force_env, "YES") == 0))
+	    || (clicolor && (strcmp(clicolor, "YES") == 0))
+	    || (colorterm && (strcmp(clicolor, "YES") == 0))) {
+		if (parse_bsd_ls_colors(colors, getenv("LSCOLORS")) != 0) {
```

Comment:
I think rather than calling this unconditionally and then relying on `parse_gnu_ls_colors` to overwrite it, it needs to be more like

```c
static bool isset(const char *str) {
    return str && *str;
}

const char *gnu_colors = getenv("LS_COLORS");
const char *bfs_colors = getenv("BFS_COLORS");
const char *bsd_colors = getenv("LSCOLORS");
if (isset(gnu_colors) || isset(bfs_colors)) {
    parse_gnu_ls_colors(colors, gnu_colors);
    parse_gnu_ls_colors(colors, bfs_colors);
} else if (isset(bsd_colors)) {
    parse_bsd_ls_colors(colors, bsd_colors);
}
```